src/caching/Arweave/ArweaveClient.ts

```diff
@@ -117,9 +122,17 @@ export class ArweaveClient {
    * they do not match the given validator.
    * @param tag The tag to filter all the transactions by
    * @param validator The validator to validate the retrieved values
+   * @param useProvidedJWTAddressForQueries If true, the client's address will be used for queries. Otherwise, the default address is used.
    * @returns The records if they exist, otherwise an empty array
    */
-  async getByTopic<T>(tag: string, validator: Struct<T>): Promise<{ data: T; hash: string }[]> {
+  async getByTopic<T>(
+    tag: string,
+    validator: Struct<T>,
+    useProvidedJWTAddressForQueries = false
```

Comment:
Why not just pass in the address and set its default to `DEFAULT_ARWEAVE_STORAGE_ADDRESS`?

Reply:
Done